src/Datadog.Trace/Tracer.cs

```diff
@@ -415,6 +427,123 @@ internal void StartDiagnosticObserversInternal()
             }
         }
 
+        internal async Task WriteDiagnosticLog()
```

Comment:
In Azure App Services we may initialize the Tracer to send to one port for the trace-agent and one port for dogstatsd, and then update it later after using the filesystem to record which ports we were able to open for those processes. Perhaps this `WriteDiagnosticLog` method should be included in the `TracingProcessManager.SubscribeToTraceAgentPortOverride` and `TracingProcessManager.SubscribeToDogStatsDPortOverride` callbacks, if that won't be too noisy.

Reply:
The issue is that we're supposed to have only one "DATADOG TRACER CONFIGURATION" log. I'll leave it as is for now.